### PR TITLE
Consistent use of left_input & right_input

### LINKED ISSUE #2112: Harmonize input_left and left_input
In the LQP, we use `left_input`, in the PQP it is `input_left`.

Any preferences?

### PR DESCRIPTION
**Pull request by mweisgut**
fixes #2112 

Replacements in all .hpp and .cpp files:
- `input_left` -> `left_input`
- `input_right` -> `right_input`
- `input_table_left` -> `left_input_table`
- `input_table_right` -> `right_input_table`

### CODE REVIEW

File: src/lib/logical_query_plan/join_node.cpp [hunk 1]
```diff
@@ -72,29 +72,29 @@ std::vector<std::shared_ptr<AbstractExpression>> JoinNode::column_expressions()
...
     return true;
   }
 
   if (join_mode == JoinMode::FullOuter) {
     return true;
   }
 
-  if (column_is_from_left_input) {
-    return left_input()->is_column_nullable(column_id);
+  if (column_is_from_input_left) {
```
Comment (mrks) [inline src/lib/logical_query_plan/join_node.cpp:92]: .

File: src/lib/logical_query_plan/lqp_utils.cpp [hunk 2]
```diff
@@ -134,12 +134,12 @@ void lqp_replace_node(const std::shared_ptr<AbstractLQPNode>& original_node,
...
-  original_node->set_left_input(nullptr);
-  original_node->set_right_input(nullptr);
+  original_node->set_input_left(nullptr);
+  original_node->set_input_right(nullptr);
 }
 
-void lqp_remove_node(const std::shared_ptr<AbstractLQPNode>& node, const AllowRightInput allow_right_input) {
-  Assert(allow_right_input == AllowRightInput::Yes || !node->right_input(),
+void lqp_remove_node(const std::shared_ptr<AbstractLQPNode>& node, const AllowInputRight allow_input_right) {
+  Assert(allow_input_right == AllowInputRight::Yes || !node->input_right(),
```
Comment (mrks) [inline src/lib/logical_query_plan/lqp_utils.cpp:142]: This sounds somewhat yodaesque. I think I'd leave it the way it was before.

File: src/lib/operators/join_hash.cpp [hunk 3]
```diff
@@ -129,12 +129,12 @@ std::shared_ptr<const Table> JoinHash::_on_execute() {
    * JoinMode::FullOuter    Not supported by JoinHash
    * JoinMode::Semi/Anti*   The left relation becomes the probe side, the right relation becomes the build side
    */
-  const auto build_hash_table_for_right_input =
+  const auto build_hash_table_for_input_right =
```
Comment (mrks) [inline src/lib/operators/join_hash.cpp:132]: .

File: src/lib/operators/union_positions.cpp [hunk 4]
```diff
@@ -78,7 +78,7 @@ std::shared_ptr<const Table> UnionPositions::_on_execute() {
     return early_result;
   }
 
-  const auto& left_input_table = *input_table_left();
+  const auto& input_left_table = *input_table_left();
```
Comment (mrks) [inline src/lib/operators/union_positions.cpp:81]: .

File: src/lib/optimizer/strategy/column_pruning_rule.cpp [hunk 5]
```diff
@@ -171,12 +171,12 @@ ExpressionUnorderedSet gather_locally_required_expressions(
     case LQPNodeType::Export:
     case LQPNodeType::Update:
     case LQPNodeType::ChangeMetaTable: {
-      const auto& left_input_expressions = node->left_input()->column_expressions();
-      locally_required_expressions.insert(left_input_expressions.begin(), left_input_expressions.end());
+      const auto& input_left_expressions = node->input_left()->column_expressions();
```
Comment (mrks) [inline src/lib/optimizer/strategy/column_pruning_rule.cpp:174]: Cases where we have `left_input_whatever` seem to be  difficult to automatically replace.

File: src/lib/optimizer/strategy/column_pruning_rule.cpp [hunk 6]
```diff
@@ -231,16 +231,16 @@ void try_join_to_semi_rewrite(
...
-  auto left_input_is_used = false;
-  auto right_input_is_used = false;
+  auto input_left_is_used = false;
+  auto input_right_is_used = false;
   for (const auto& output : node->outputs()) {
     for (const auto& required_expression : required_expressions_by_node.at(output)) {
-      if (expression_evaluable_on_lqp(required_expression, *node->left_input())) left_input_is_used = true;
-      if (expression_evaluable_on_lqp(required_expression, *node->right_input())) right_input_is_used = true;
+      if (expression_evaluable_on_lqp(required_expression, *node->input_left())) input_left_is_used = true;
+      if (expression_evaluable_on_lqp(required_expression, *node->input_right())) input_right_is_used = true;
```
Comment (mrks) [inline src/lib/optimizer/strategy/column_pruning_rule.cpp:239]: Here it is ok

File: src/lib/sql/sql_translator.cpp [hunk 7]
```diff
@@ -792,9 +792,9 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_predicated_join(const
   auto join_predicates = std::vector<std::shared_ptr<AbstractExpression>>{};
 
   for (const auto& predicate : raw_join_predicate_cnf) {
-    if (expression_evaluable_on_lqp(predicate, *left_input_lqp)) {
+    if (expression_evaluable_on_lqp(predicate, *input_left_lqp)) {
```
Comment (mrks) [inline src/lib/sql/sql_translator.cpp:795]: input_lqp_left?

File: src/lib/statistics/cardinality_estimator.cpp [hunk 8]
```diff
@@ -117,29 +117,29 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_statistics(
    * 2. Cache lookup failed - perform an actual cardinality estimation
    */
   auto output_table_statistics = std::shared_ptr<TableStatistics>{};
-  const auto left_input_table_statistics = lqp->left_input() ? estimate_statistics(lqp->left_input()) : nullptr;
-  const auto right_input_table_statistics = lqp->right_input() ? estimate_statistics(lqp->right_input()) : nullptr;
+  const auto input_left_table_statistics = lqp->input_left() ? estimate_statistics(lqp->input_left()) : nullptr;
```
Comment (mrks) [inline src/lib/statistics/cardinality_estimator.cpp:120]: Really weird

File: src/lib/logical_query_plan/lqp_utils.cpp (hunk 2 above)
Comment (Bouncner) [reply src/lib/logical_query_plan/lqp_utils.cpp:142]: +1 for `yodaesque`. :D

**Comment by mweisgut** [comment]
left_input_something identifiers:
```
input_left_adapted
input_left_base_column_statistics
input_left_column_count
input_left_column_statistics
input_left_expression
input_left_expressions
input_left_is_used
input_left_lqp
input_left_operator
input_left_row_count
input_left_row_set
input_left_table
input_left_table_statistics
input_left_table_type
```

something_input_left identifiers:
```
_input_left
alias_node_other_input_left
column_is_from_input_left
copied_input_left
mutable_input_left
set_input_left
```

right_input_something identifiers:
```
input_right_adapted
input_right_column_statistics
input_right_expression
input_right_expressions
input_right_is_used
input_right_lqp
input_right_operator
input_right_row_count
input_right_row_set
input_right_table_statistics
input_right_table_type
```

something_input_right identifiers:
```
_input_right
allow_input_right
build_hash_table_for_input_right
copied_input_right
mutable_input_right
set_input_right
```

In all these cases the left/right_input version sounds more like English natural language. I suggest to stick with this pattern then and also use `right_input` and `left_input`. What do you think?

**Comment by mrks** [comment]
> I propose to stick with this pattern then and also use right_input and left_input. What do you think?

Are you suggesting that, different from what was decided in the fortnightly, we go for `left_input` instead of `input_left`? If yes, fine with me - most examples sound better that way. In that case, we should ping a couple of people (including those who voted for `input_left`) to get their feedback.

**Comment by mweisgut** [comment]
> > I propose to stick with this pattern then and also use right_input and left_input. What do you think?
> 
> Are you suggesting that, different from what was decided in the fortnightly, we go for `left_input` instead of `input_left`? If yes, fine with me - most examples sound better that way. In that case, we should ping a couple of people (including those who voted for `input_left`) to get their feedback.

Yes, that's what I meant.
@Bensk1 @Bouncner @julianmenzler

**Comment by Bouncner** [comment]
Haven't thought about these cases. I am fine with `left_input`.

**Comment by Bensk1** [comment]
I would be fine with both.

**Comment by julianmenzler** [comment]
> Haven't thought about these cases. I am fine with `left_input`.

Me neither. So let's go for `left_input` / `right_input` then.

**Comment by mweisgut** [comment]
Related software engineering SE question/discussion: https://softwareengineering.stackexchange.com/questions/347944/naming-procedure-conventions-where-to-place-adjective-in-a-verb-followed-by-an

**Pull request merged**